### PR TITLE
REDSHOP-5888 [Bug] Error "missing-input-response" in registration user

### PR DESCRIPTION
**Pull request by NguyenBao10**

### CODE REVIEW

File: component/site/models/registration.php [hunk 1]
```diff
@@ -35,7 +35,21 @@ public function __construct()
 
 	public function store(&$data)
 	{
-		$captcha = Redshop\Helper\Utility::checkCaptcha($data);
+        $plugin = JPluginHelper::getPlugin('captcha','recaptcha');
```
Comment (anhredweb) [inline component/site/models/registration.php]: @NguyenBao10 : please add **\** as best practice

File: component/site/models/registration.php [hunk 2]
```diff
@@ -35,7 +35,21 @@ public function __construct()
...
+
+        $dataCaptcha = $data;
+
+        if ($plugin)
+        {
+            $params  = new JRegistry($plugin->params);
```
Comment (anhredweb) [inline component/site/models/registration.php]: @NguyenBao10 `\JRegistry` or even better `\Joomla\Registry\Registry`

File: component/site/models/registration.php [hunk 3]
```diff
@@ -35,7 +35,21 @@ public function __construct()
...
+
+            if ($params->get('version', '') === '2.0')
```
Comment (anhredweb) [inline component/site/models/registration.php]: @NguyenBao10 what happen if version of plugin is `2.0.1`?

File: component/site/models/registration.php [hunk 4]
```diff
@@ -35,7 +35,21 @@ public function __construct()
...
+            {
+                $dataCaptcha = null;
+            }
+        }
+
+		$captcha = Redshop\Helper\Utility::checkCaptcha($dataCaptcha);
```
Comment (anhredweb) [inline component/site/models/registration.php]: @NguyenBao10 same here, please add `\`

File: component/site/models/registration.php (hunk 3 above)
Comment (NguyenBao10) [reply component/site/models/registration.php]: its version of recapthca and at the current, it just has 2 fixed option (1.0 and 2.0). If recaptcha releases a new version in the future, we will check the plugin again and fix later

**Comment by NguyenBao10** [comment]
updated a @anhredweb

**Pull request merged**